backend/repository/manager.go

```diff
@@ -229,15 +244,25 @@ func (m *Manager) UpdateEndpoints(r *Repository, endpointCfgDir string, req []En
 	return nil
 }
 
+// ManagedThriftFile represents a thrift file we want to write
+type ManagedThriftFile struct {
+	SourceCode string `json:"source_code"`
+	Filename   string `json:"filename"`
+}
+
 // UpdateRequest is the request to update thrift files, clients and endpoints.
 type UpdateRequest struct {
-	ThriftFiles     []string         `json:"thrift_files"`
-	ClientUpdates   []ClientConfig   `json:"client_updates"`
-	EndpointUpdates []EndpointConfig `json:"endpoint_updates"`
+	ThriftFiles        []string            `json:"thrift_files"`
+	ManagedThriftFiles []ManagedThriftFile `json:"managed_thrift_files"`
+	ClientUpdates      []ClientConfig      `json:"client_updates"`
+	EndpointUpdates    []EndpointConfig    `json:"endpoint_updates"`
 }
 
 // UpdateAll writes the updates for thrift files, clients and endpoints.
 func (m *Manager) UpdateAll(r *Repository, clientCfgDir, endpointCfgDir string, req *UpdateRequest) error {
+	if err := m.UpdateManagedThriftFiles(r, req.ManagedThriftFiles); err != nil {
```

Comment:
it would be nice to consolidate this with `UpdateThriftFiles`
thrift either from fetching idl `m.thriftMetaInIDLRegistry(paths)`, or direct user input, the rest of the code path could be reuse? (dependencies ... etc)

what if I put a `include` from idl in my input thrift?

Reply:
I plan to add include support in a seperate PR. the way include support will work is slightly different because this file that is written does not yet exist in `idl-registry` so leveraging `idl-registry` code is a bit harder.